agent/task_context.go

```diff
@@ -348,9 +350,9 @@ func (tc *taskContext) getTimeout(taskGroup string) (*commandBlock, error) {
 
 	return &commandBlock{
 		block:       command.TaskTimeoutBlock,
-		commands:    tg.Timeout,
+		commands:    tc.taskConfig.TaskGroup.Timeout,
 		timeoutKind: callbackTimeout,
-		getTimeout:  tc.getTaskGroupCallbackTimeout(tg),
+		getTimeout:  tc.getTaskGroupCallbackTimeout(&tc.taskConfig.TaskGroup),
```

Comment:
It seems like these functions don't need to pass in the `tc.taskConfig.TaskGroup` (and in the other places in this file) because `tc.getTaskGroupCallbackTimeout` already has the task group available to check.